File: 1/2021.12.20/475.Heaters/java/solution.java
class Solution {
    public int findRadius(int[] houses, int[] heaters) {
        Arrays.sort(houses);
        Arrays.sort(heaters);
        int res = 0;
        int j = 0;
        for (int i = 0; i < houses.length; ++i) {
            int curDistance = Math.abs(houses[i] - heaters[j]);
            while (j + 1 < heaters.length && Math.abs(houses[i] - heaters[j + 1]) <= curDistance) {
                ++j;
                curDistance = Math.abs(houses[i] - heaters[j]);
            }
            res = Math.max(res, curDistance);
        }
        return res;
    }
}

File: 1/2021.12.20/475.Heaters/python/solution.py
class Solution:
    def findRadius(self, houses: List[int], heaters: List[int]) -> int:
        houses.sort()
        heaters.sort()
        res = 0
        j = 0
        for i in range(len(houses)):
            cur = abs(houses[i] - heaters[j])
            while j < len(heaters) - 1 and abs(houses[i] - heaters[j]) >= abs(houses[i] - heaters[j + 1]):
                j += 1
                cur = min(cur, abs(houses[i] - heaters[j]))
            res = max(res, cur)
        return res
